cocos/core/animation/cross-fade.ts

```diff
@@ -76,6 +86,10 @@ export class CrossFade extends Playable {
                 state.sample();
             }
         }
+
+        if (managedStates.length === 1 && fadings.length === 1) { // Definitely not code repetition
```

Comment:
If cross fade has finished its work, unschedule itself.

Reply:
I have documented the `Definitely not code repetition` since both lengths of `managedStates` and `fadings` might varies during `this._updateWeights()`